common/djangoapps/track/backends/mongodb.py

```diff
@@ -67,7 +70,7 @@ def __init__(self, **kwargs):
         database = self.connection[db_name]
 
         if user or password:
-            database.authenticate(user, password)
+            database.authenticate(user, password, auth_source=auth_source)
```

Comment:
@xitij2000, is `auth_source` the correct keyword argument name? The PyMongo 2.9.1 docs [here](http://api.mongodb.com/python/2.9.1/examples/authentication.html#delegated-authentication) mention that `source` is the keyword parameter name. I checked the PyMongo source [here](https://github.com/mongodb/mongo-python-driver/blob/2.9.1/pymongo/database.py#L1071) and while `source` is a known argument, `auth_source` is treated as a part of arbitrary arguments `**kwargs` and validated against a known set of authentication options in which `auth_source` is not present. Even in the change below, the `.authenticate()` method called on the database object uses the `source` argument.

Am I missing something obvious here?

I also checked what happens when this new parameter is unset (which is the current case) and since the truthiness is checked [here](https://github.com/mongodb/mongo-python-driver/blob/2.9.1/pymongo/database.py#L1076), I guess everything should work fine.

Reply:
@lgp171188 Yup, I had fixed this in the upstream PR, but forgot to cherry-pick it here. Will fix.

Reply:
@xitij2000, let me know once you have cherry-picked it.

Reply:
Done.